mediapipe/examples/android/src/java/com/google/mediapipe/apps/instantmotiontracking/MainActivity.java

```diff
@@ -17,107 +17,306 @@
+          getPackageManager().getApplicationInfo(getPackageName(), PackageManager.GET_META_DATA);
     } catch (NameNotFoundException e) {
       Log.e(TAG, "Cannot find application info: " + e);
     }
 
     previewDisplayView = new SurfaceView(this);
-
     setupPreviewDisplayView();
 
     // Initialize asset manager so that MediaPipe native libraries can access the app assets, e.g.,
     // binary graphs.
     AndroidAssetUtil.initializeNativeAssetManager(this);
     eglManager = new EglManager(null);
     processor =
-            new FrameProcessor(
-                    this,
-                    eglManager.getNativeContext(),
-                    applicationInfo.metaData.getString("binaryGraphName"),
-                    applicationInfo.metaData.getString("inputVideoStreamName"),
-                    applicationInfo.metaData.getString("outputVideoStreamName"));
+        new FrameProcessor(
+            this,
+            eglManager.getNativeContext(),
+            applicationInfo.metaData.getString("binaryGraphName"),
+            applicationInfo.metaData.getString("inputVideoStreamName"),
+            applicationInfo.metaData.getString("outputVideoStreamName"));
     processor.getVideoSurfaceOutput().setFlipY(FLIP_FRAMES_VERTICALLY);
+    PermissionHelper.checkAndRequestCameraPermissions(this);
 
-	//AndroidPacketCreator packetCreator = processor.getPacketCreator();
-    //Map<String, Packet> inputSidePackets = new HashMap<>();
-    //inputSidePackets.put("inputSelectedAnchorName", packetCreator.createInt32Array(new int[]{50,50,50,50,50,50,50,50,50,50,50,50,50,50,50,50,50,50,50,50,50,50,50,50}));
-    //processor.setInputSidePackets(inputSidePackets);
+    // Send loaded 3d render assets as side packets to graph
+    prepareDemoAssets();
+    AndroidPacketCreator packetCreator = processor.getPacketCreator();
+    Map<String, Packet> inputSidePackets = new HashMap<>();
+    inputSidePackets.put("obj_asset_name", packetCreator.createString(OBJ_FILE));
+    inputSidePackets.put("box_asset_name", packetCreator.createString(BOX_FILE));
+    inputSidePackets.put("obj_texture", packetCreator.createRgbaImageFrame(objTexture));
+    inputSidePackets.put("box_texture", packetCreator.createRgbaImageFrame(boxTexture));
+    processor.setInputSidePackets(inputSidePackets);
 
-    PermissionHelper.checkAndRequestCameraPermissions(this);
+    // Add frame listener to PacketManagement system
+    mediaPipePacketManager = new MediaPipePacketManager();
+    processor.setOnWillAddFrameListener(mediaPipePacketManager);
+
+    // Begin with 0 stickers in dataset
+    stickerList = new ArrayList<Sticker>();
+    currentSticker = null;
+
+    // Define sensor properties (only get one orientation sensor)
+    SensorManager sensorManager = (SensorManager) getSystemService(SENSOR_SERVICE);
+    List sensorList = sensorManager.getSensorList(Sensor.TYPE_ORIENTATION);
```

Comment:
This is deprecated. Suggest to use https://developer.android.com/reference/android/hardware/SensorManager#getOrientation(float[],%20float[])

Also can you use either rotation matrix instead of the three angles? This will avoid gimbal lock.